module/zfs/dsl_crypt.c

```diff
@@ -122,7 +122,7 @@ dsl_wrapping_key_create(uint8_t *wkeydata, zfs_keyformat_t keyformat,
 	/* allocate and initialize the underlying crypto key */
 	wkey->wk_key.ck_data = kmem_alloc(WRAPPING_KEY_LEN, KM_SLEEP);
 	if (!wkey->wk_key.ck_data) {
-		ret = ENOMEM;
+		ret = SET_ERROR(ENOMEM);
```

Comment:
This whole branch actually doesn't make sense, since the wait arg is `KM_SLEEP`.  It should just be removed.

Reply:
@wca those sorts of mistakes are pervasive throughout dsl_crypt.c

Reply:
```
index 0635cc3c9..11e793240 100644
--- a/module/zfs/dsl_crypt.c
+++ b/module/zfs/dsl_crypt.c
@@ -116,15 +116,9 @@ dsl_wrapping_key_create(uint8_t *wkeydata, zfs_keyformat_t keyformat,
 
        /* allocate the wrapping key */
        wkey = kmem_alloc(sizeof (dsl_wrapping_key_t), KM_SLEEP);
-       if (!wkey)
-               return (SET_ERROR(ENOMEM));
 
        /* allocate and initialize the underlying crypto key */
        wkey->wk_key.ck_data = kmem_alloc(WRAPPING_KEY_LEN, KM_SLEEP);
-       if (!wkey->wk_key.ck_data) {
-               ret = SET_ERROR(ENOMEM);
-               goto error;
-       }
 
        wkey->wk_key.ck_format = CRYPTO_KEY_RAW;
        wkey->wk_key.ck_length = CRYPTO_BYTES2BITS(WRAPPING_KEY_LEN);
@@ -138,12 +132,6 @@ dsl_wrapping_key_create(uint8_t *wkeydata, zfs_keyformat_t keyformat,
 
        *wkey_out = wkey;
        return (0);
-
-error:
-       dsl_wrapping_key_free(wkey);
-
-       *wkey_out = NULL;
-       return (ret);
 }
 
 int
@@ -161,11 +149,6 @@ dsl_crypto_params_create_nvlist(dcp_cmd_t cmd, nvlist_t *props,
        char *keylocation = NULL;
 
        dcp = kmem_zalloc(sizeof (dsl_crypto_params_t), KM_SLEEP);
-       if (!dcp) {
-               ret = SET_ERROR(ENOMEM);
-               goto error;
-       }
-
        dcp->cp_cmd = cmd;
 
        /* get relevant arguments from the nvlists */
@@ -561,8 +544,6 @@ dsl_crypto_key_open(objset_t *mos, dsl_wrapping_key_t *wkey,
 
        /* allocate and initialize the key */
        dck = kmem_zalloc(sizeof (dsl_crypto_key_t), KM_SLEEP);
-       if (!dck)
-               return (SET_ERROR(ENOMEM));
 
        /* fetch all of the values we need from the ZAP */
        ret = zap_lookup(mos, dckobj, DSL_CRYPTO_KEY_CRYPTO_SUITE, 8, 1,
@@ -2430,9 +2411,7 @@ dsl_crypto_populate_key_nvlist(dsl_dataset_t *ds, uint64_t from_ivset_guid,
        VERIFY0(dmu_objset_from_ds(ds, &os));
        mdn = DMU_META_DNODE(os);
 
-       ret = nvlist_alloc(&nvl, NV_UNIQUE_NAME, KM_SLEEP);
-       if (ret != 0)
-               goto error;
+       nvlist_alloc(&nvl, NV_UNIQUE_NAME, KM_SLEEP);
 
        /* lookup values from the DSL Crypto Key */
        ret = zap_lookup(mos, dckobj, DSL_CRYPTO_KEY_CRYPTO_SUITE, 8, 1,
```
orthogonal to current work